### PR TITLE
2064 - New wizard form

### PR DESCRIPTION
**Pull request by Cooryd**
#### Background

Still to do:

- [ ] Look into why whole step is re-rendering multiple times onFocus. (Probably because we pass all formic props down)
- [x] Unit test for flattenObject
- [ ] Next button on disclosure page needs to trigger modal and call submit.
- [ ] Implement auto save again
- [ ] Pass correct props down to file step for uploading
- [x] Investigate capitalised url path
- [ ] check validation error message persisting after adding the two required editors/ reviewers

#### Any relevant tickets

Please use the phrase 'Closes' if it will fully complete the ticket.

#2064 

#### How has this been tested?
**Reviewers** ensure that these test requirements are also reviewed.

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. 

- [ ] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

#### UI Changes

- [ ] Has been reviewed against Designs
- [ ] Necessary updates to styleguide

### CODE REVIEW

**Review by diversemix — APPROVED** [review]
On the understanding that deployment to Production is turned off and we need to do the other associated tickets

File: packages/component-submission/client/pages/FilesStepPage.js [hunk 1]
```diff
@@ -213,3 +213,5 @@ export default compose(
     manuscriptUploadProgress: props.uploadProgress.manuscriptUploadProgress,
   })),
 )(FilesStepPageComponent)
+
+export const fields = ['coverLetter', 'files', 'fileStatus']
```
Comment (Cooryd) [inline packages/component-submission/client/pages/FilesStepPage.js]: remove this.

File: packages/component-submission/client/pages/NewSubmissionWizard.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  editorsSchema,
+} from '../utils/ValidationSchemas'
+
+const BoxNoMinWidth = styled(Box)`
+  min-width: 0;
+`
+
+const NewSubmissionWizard = ({ initialValues, match, history }) => {
+  const getCurrentStepFromPath = () =>
+    STEP_NAMES.map(step => step.toLowerCase()).indexOf(
```
Comment (Cooryd) [inline packages/component-submission/client/pages/NewSubmissionWizard.js:38]: can we remove the `toLowerCase` here?

File: packages/component-submission/client/pages/NewSubmissionWizard.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                  from="/newSubmit/:id"
+                  to={`/newSubmit/${match.params.id}/author`}
+                />
+                <ErrorPage error="404: page not found" />
+              </Switch>
+              <Flex mt={6}>
+                <Box mr={3}>
+                  <Button
+                    data-test-id="back"
+                    disabled={currentStep === 0}
```
Comment (Cooryd) [inline packages/component-submission/client/pages/NewSubmissionWizard.js:93]: raise issue for hiding this on first step

**Review by erezmus — APPROVED** [review]

**Pull request merged**